Fascinating that the trial psycopg test had a fluke failure:

```
[FAIL]
Traceback (most recent call last):
  File "/home/runner/work/synapse/synapse/tests/rest/admin/test_user.py", line 5785, in test_redact_messages_all_rooms
    self.fail("Timed out waiting for redactions.")
twisted.trial.unittest.FailTest: Timed out waiting for redactions.

tests.rest.admin.test_user.UserRedactionBackgroundTaskTestCase.test_redact_messages_all_rooms
```

It appears this test is utilizing `time.time()` to check the timeout. Unfortunately, this is considered a flake as using fake time and real time in a test is automatically suspect when any kind of contention is happening(*and* is beyond the scope of what we are trying to accomplish)

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -692,6 +692,9 @@ jobs:
           - arrangement: monolith
             database: Psycopg
 
+          - arrangement: workers
+            database: Psycopg
+
     steps:
       - name: Checkout synapse codebase
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2

--- a/scripts-dev/complement.sh
+++ b/scripts-dev/complement.sh
@@ -271,8 +271,12 @@ main() {
     export PASS_SYNAPSE_WORKER_TYPES="$WORKER_TYPES"
 
     # Workers can only use Postgres as a database.
-    export PASS_SYNAPSE_COMPLEMENT_DATABASE=postgres
-
+    # A handy pattern for lower-casing all letters in a variable, `${variable,,}`
+    if [[ "${POSTGRES,,}" = "psycopg" ]]; then
+      export PASS_SYNAPSE_COMPLEMENT_DATABASE=psycopg
+    else
+      export PASS_SYNAPSE_COMPLEMENT_DATABASE=postgres
+    fi
     # And provide some more configuration to complement.
 
     # It can take quite a while to spin up a worker-mode Synapse for the first

